### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by GnomedDev**
RA flagged all these warnings, so I just fixed all of them.

### CODE REVIEW

File: src/read/unit.rs [hunk 1]
```diff
@@ -4207,11 +4207,7 @@ mod tests {
 
     #[test]
     fn test_attribute_udata_sdata_value() {
-        let tests: &[(
-            AttributeValue<EndianSlice<LittleEndian>>,
-            Option<u64>,
-            Option<i64>,
-        )] = &[
+        let tests: &[(AttributeValue<EndianSlice<LittleEndian>>, _, _)] = &[
```
Comment (philipc) [inline src/read/unit.rs:4210]: I think this is better left how it is. I assume this is `clippy::type_complexity`, but hiding complexity doesn't make it better.

File: src/write/op.rs [hunk 2]
```diff
@@ -1100,417 +1100,417 @@ mod tests {
...
-                                read::Operation::WasmGlobal { index: 1000 },
-                            ),
-                            (
-                                &|x| x.op_wasm_stack(1000),
-                                Operation::WasmStack(1000),
-                                read::Operation::WasmStack { index: 1000 },
-                            ),
-                        ];
+                    #[allow(clippy::type_complexity)]
+                    let operations: &[(&dyn Fn(&mut Expression), _, _)] = &[
```
Comment (philipc) [inline src/write/op.rs:1104]: I assume this change was an attempt to fix `clippy::type_complexity`? I prefer not to do it.

File: src/read/cfi.rs [hunk 3]
```diff
@@ -6718,15 +6718,16 @@ mod tests {
         let section = EndianSlice::new(&section, LittleEndian);
 
         let mut offset = None;
-        match parse_fde(
+        let result = parse_fde(
```
Comment (philipc) [inline src/read/cfi.rs:6721]: Which warning is this?

Comment (GnomedDev) [reply src/read/cfi.rs:6721]: clippy::blocks_in_conditions

**Review by philipc — APPROVED** [review]

**Pull request merged**